profiler/src/ProfilerEngine/Datadog.Linux.ApiWrapper/functions_to_wrap.c

```diff
@@ -59,13 +60,210 @@ unsigned long long dd_inside_wrapped_functions()
     return functions_entered_counter;
 }
 
+#if defined(__aarch64__)
+const char* DdDotnetFolder = "linux-arm64";
+const char* DdDotnetMuslFolder = "linux-musl-arm64";
+#else
+const char* DdDotnetFolder = "linux-x64";
+const char* DdDotnetMuslFolder = "linux-musl-x64";
+#endif
+
+char* crashHandler = NULL;
+
+const char* getLibraryPath()
+{
+    Dl_info dl_info;
+
+    if (dladdr((void*)getLibraryPath, &dl_info))
+    {
+        return dl_info.dli_fname;
+    }
+    
+    return NULL;
+}
+
+int isAlpine()
+{
+    if (access("/etc/alpine-release", F_OK) == 0)
+    {
+        return 1;
+    }
+
+    return 0;
+}
+
+char* appendToPath(const char* folder, const char* suffix)
+{
+    int length = strlen(folder) + strlen(suffix);
+
+    char* result = malloc(length + 1);
+
+    if (result == NULL)
+    {
+        return NULL;
+    }
+
+    strcpy(result, folder);
+    strcpy(result + strlen(folder), suffix);
+
+    return result;
+}
+
+char* getFolder(const char* path)
+{
+    char* lastSlash = strrchr(path, '/');
+    if (lastSlash == NULL)
+    {
+        return "";
+    }
+
+    // Copying the last subfolder to a new string
+    int length = lastSlash - path;
+    char* folder = malloc(length + 1);
+
+    if (folder == NULL)
+    {
+        return NULL; // Memory allocation failed
+    }
+
+    strncpy(folder, path, length);
+    folder[length] = '\0';
+
+    return folder;
+}
+
+char* getSubfolder(const char* path)
+{
+    char* lastSlash = strrchr(path, '/');
+    if (lastSlash == NULL) {
+        return NULL; // No slash found, path does not contain subfolders
+    }
+
+    // Finding the last but one slash
+    char* temp = lastSlash - 1;
+    while (temp >= path && *temp != '/') {
+        temp--;
+    }
+
+    if (temp < path)
+    {
+        // No leading slash, that's weird
+        return NULL;
+    }
+
+    // Increment to move past the slash
+    temp++;
+
+    // Copying the last subfolder to a new string
+    int folderLength = lastSlash - temp;
```

Comment:
Again, if there's a `//`, then `folderLength` could be 0 I think? Worth handling explicitly?

Reply:
I think that works even if length is 0 🤔